ext/date/php_date.c

```diff
@@ -4108,39 +4109,58 @@ static int date_period_initialize(timelib_time **st, timelib_time **et, timelib_
 /* {{{ Creates new DatePeriod object. */
 PHP_METHOD(DatePeriod, __construct)
 {
```

Comment:
I'm not convinced by the changes here. This is a heavily overloaded signature, and I think the current approach of treating this as three different possible signatures makes more sense than taking the union of the signatures and combining it into one.

Reply:
@derickr Any opinion about this change?

I love that the nested ZPPs goes away + we'll have correct type info, but I can also understand if we should opt for the overloaded signature. Although, In my opinion the error messages are more helpful in case of the new approach than just saying which combinations of types are accepted.

Reply:
After a short discussion in chat with Derick, I reverted the ZPP changes. Derick's preference is to deprecate this constructor, and introduce 3 factory methods. I pretty much support this so I'll add it to our list of candidates for deprecation.